### PR TITLE
allow global route to be specified using "*"

### PR DESCRIPTION
**Pull request by elerch**
Note that this can be done for all methods or just a single method.

FYI - this is literally my first code in zig, so I'm all ears if there's a better way. ;) I needed this for a project and thought it would be useful.

### CODE REVIEW

File: src/routez/router.zig [hunk 1]
```diff
@@ -121,6 +121,10 @@ pub fn match(
     var len: usize = undefined;
 
     inline for (route.path) |c, i| {
+        comptime if (state == .Start and c == '*') {
```
Comment (Vexu) [inline src/routez/router.zig:124]: Move this inside the switch in the `.Start` after `'/'`.

File: src/routez/router.zig [hunk 2]
```diff
@@ -302,7 +306,7 @@ pub fn match(
         }
     };
     const r = pathbuf[begin..index];
-    if (!mem.eql(u8, r, path[path_index..])) {
+    if (!mem.eql(u8, r, path[path_index..]) and route.path[0] != '*') {
```
Comment (Vexu) [inline src/routez/router.zig:309]: Put this in the other order so that the if will not be emitted for global paths.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: src/routez/router.zig (hunk 1 above)
Comment (elerch) [reply src/routez/router.zig:124]: I thought I tried that and had some issue. I'll give it a another shot.

Comment (elerch) [reply src/routez/router.zig:124]: Right - this is my knowledge of the language (or lack thereof). I wanted to bail out at start if it saw the '*' character, but break inside a switch I think just breaks out of the switch, not the whole for loop.

Comment (Vexu) [reply src/routez/router.zig:124]: Breaking out of a switch is not a thing in Zig, you would need to use a labeled block for that.

**Comment by elerch** [comment]
I made the changes, but I noticed in the process of testing that there is an existing issue in routez. Even without this PR applied, the following will not work as one would expect:

```zig
pub fn main() !void {
    var server = Server.init(
        allocator,
        .{},
        .{
            get("/about", aboutHandler),
            post("/about", aboutPostHandler),
        },
    );
    var addr = try Address.parseIp("127.0.0.1", 8081);
    try server.listen(addr);
}
```

If I'm reading the code correctly, this is due to router.zig line 316 (which you've already noted in your comment is a todo). What I haven't worked out is that when I for instance use the example and issue `curl -IX POST http://localhost:8081/about/more` I don't get a 405 MethodNotAllowed as it looks like I should be based on the source. Rather, I get:

```
HTTP/1.1 200 OK
connection: close
content-length: 0
```

**Comment by Vexu** [comment]
Thanks!

>If I'm reading the code correctly, this is due to router.zig line 316 (which you've already noted in your comment is a todo). What I haven't worked out is that when I for instance use the example and issue `curl -IX POST http://localhost:8080/about/more` I don't get a 405 MethodNotAllowed as it looks like I should be based on the source. Rather, I get:
```
HTTP/1.1 200 OK
connection: close
content-length: 0
```

I don't know if your example is incomplete but when I do `curl -IX POST http://localhost:8080/about/more` I get the 404 page.

**Pull request merged**